beacon-chain/state/state-native/setters_misc.go

```diff
@@ -198,9 +199,14 @@ func (b *BeaconState) addDirtyIndices(index types.FieldIndex, indices []uint64)
 		return
 	}
 	totalIndicesLen := len(b.dirtyIndices[index]) + len(indices)
+	// Reduce duplicates to verify that these are indeed unique.
+	if totalIndicesLen > indicesLimit {
+		b.dirtyIndices[index] = slice.SetUint64(b.dirtyIndices[index])
+	}
+	totalIndicesLen = len(b.dirtyIndices[index]) + len(indices)
```

Comment:
No need to recompute this if the above if wasn't hit, so please move this up within the if branch